synapse/lib/view.py

```diff
@@ -138,12 +176,264 @@ async def __anit__(self, core, node):
         self.invalid = None
         self.parent = None  # The view this view was forked from
 
+        # This will be True for a view which has a merge in progress
+        self.merging = self.info.get('merging', False)
+
         # isolate some initialization to easily override.
         await self._initViewLayers()
 
         self.trigtask = None
         await self.initTrigTask()
 
+        self.mergetask = None
+
+    def reqParentQuorum(self):
+
+        if self.parent is None:
+            mesg = f'View ({self.iden}) has no parent.'
+            raise s_exc.SynErr(mesg=mesg)
+
+        quorum = self.parent.info.get('quorum')
+        if quorum is None:
+            mesg = f'Parent view of ({self.iden}) does not require quorum voting.'
+            raise s_exc.SynErr(mesg=mesg)
+
+        return quorum
+
+    def reqNoParentQuorum(self):
+
+        if self.parent is None:
+            return
+
+        quorum = self.parent.info.get('quorum')
+        if quorum is not None:
+            mesg = f'Parent view of ({self.iden}) requires quorum voting.'
+            raise s_exc.SynErr(mesg=mesg)
+
+    async def _wipeViewMeta(self):
+        for lkey in self.core.slab.scanKeysByPref(self.bidn, db='view:meta'):
+            self.core.slab.delete(lkey, db='view:meta')
+            await asyncio.sleep(0)
+
+    def getMergeRequest(self):
+        byts = self.core.slab.get(self.bidn + b'merge:req', db='view:meta')
```

Comment:
before we've got this in the weeds do we want to add a null byte between the iden and name so we can easily split lkeys and + iterate the meta db in the future?

Reply:
I don't think that's possible here. `bidn` could contain a null byte which would break splitting.

Reply:
Yea, these fields are split based on fixed size values so they don't require split chars :+1: